internal/bminventory/inventory.go

```diff
@@ -4048,6 +4054,35 @@ func (b *bareMetalInventory) validateClusterInfraEnvRegister(ctx context.Context
 	return nil
 }
 
+func (b *bareMetalInventory) setIgnitionConfigOverrideUsage(clusterId strfmt.UUID, ignitionConfigOverride, level string, db *gorm.DB) error {
+	var cluster *common.Cluster
+	cluster, err := common.GetClusterFromDBForUpdate(db, clusterId, common.SkipEagerLoading)
```

Comment:
FOR UPDATE has to happen as part of a transaction, but from a quick scan of the code I don't think this instance of `db` is a transaction.

Reply:
Oh I see, right it isn't!

Reply:
It is now! All calls to this function are now transactions